src/comm/UDPLink.cc

```diff
@@ -136,36 +136,35 @@ void UDPLink::removeHost(const QString& host)
 
 void UDPLink::_writeBytes(const QByteArray data)
 {
-    if (!_socket)
-        return;
-
-    QStringList goneHosts;
-    // Send to all connected systems
-    QString host;
-    int port;
-    if(_udpConfig->firstHost(host, port)) {
-        do {
-            QHostAddress currentHost(host);
-            if(_socket->writeDatagram(data, currentHost, (quint16)port) < 0) {
-                // This host is gone. Add to list to be removed
-                // We should keep track of hosts that were manually added (static) and
-                // hosts that were added because we heard from them (dynamic). Only
-                // dynamic hosts should be removed and even then, after a few tries, not
-                // the first failure. In the mean time, we don't remove anything.
-                if(REMOVE_GONE_HOSTS) {
-                    goneHosts.append(host);
+    if (_socket) {
+        QStringList goneHosts;
+        // Send to all connected systems
+        QString host;
+        int port;
+        if(_udpConfig->firstHost(host, port)) {
+            do {
+                QHostAddress currentHost(host);
+                if(_socket->writeDatagram(data, currentHost, (quint16)port) < 0) {
+                    // This host is gone. Add to list to be removed
+                    // We should keep track of hosts that were manually added (static) and
+                    // hosts that were added because we heard from them (dynamic). Only
+                    // dynamic hosts should be removed and even then, after a few tries, not
+                    // the first failure. In the mean time, we don't remove anything.
+                    if(REMOVE_GONE_HOSTS) {
```

Comment:
This is not right. A write error simply means the TCP stack had issues sending the datagram. There is no relationship with a  "gone" host as there is no feed back if the target ever received the datagram.